domehunter/dome_control.py

```diff
@@ -0,0 +1,569 @@
+        self.current_direction = direction
+        # set the direction relay switch to CW position
+        if self.current_direction.name == "CW":
+            self.__direction_relay.on()
+            self._change_led_state(1, leds=[LED_light.RELAY_2_NO])
+            self._change_led_state(0, leds=[LED_light.RELAY_2_NC])
+        elif self.current_direction.name == "CCW":
+            self.__direction_relay.off()
+            self._change_led_state(0, leds=[LED_light.RELAY_2_NO])
+            self._change_led_state(1, leds=[LED_light.RELAY_2_NC])
+        # turn on rotation
+        self.__rotation_relay.on()
+        # update the rotation relay debug LEDs
+        self._change_led_state(1, leds=[LED_light.RELAY_1_NO])
+        self._change_led_state(0, leds=[LED_light.RELAY_1_NC])
+        cmd_status = True
+        return cmd_status
+
+    def _stop_moving(self):
+        """
+        Stop dome movement by switching the dome rotation relay off.
+        """
+        self.__rotation_relay.off()
+        # update the debug LEDs
+        self._change_led_state(0, leds=[LED_light.RELAY_1_NO])
+        self._change_led_state(1, leds=[LED_light.RELAY_1_NC])
+        # update last_direction with current_direction at time of method call
+        self.last_direction = self.current_direction
+
+    def _simulate_ticks(self, num_ticks):
+        """
+        Method to simulate encoder ticks while in testing mode.
+        """
+        tick_count = 0
+        # repeat this loop of driving the mock pins low then high to simulate
+        # an encoder tick. Continue until desired number of ticks is reached.
+        while tick_count < num_ticks:
+            self.__encoder_pin.drive_low()
+            # test_mode_delay_duration is set so that it will always exceed
+            # the set bounce_time
+            # of the pins
+            time.sleep(self.test_mode_delay_duration)
+            self.__encoder_pin.drive_high()
+            time.sleep(self.test_mode_delay_duration)
+            tick_count += 1
+
+    def _change_led_state(self, desired_state, leds=[]):  # pragma: no cover
+        """
+        Method of turning a set of debugging LEDs on
+
+        Parameters
+        ----------
+        desired_state : bool
+            Parameter to indicate whether leds are being turned on (1 or True)
+            or if they are being turned off (0 or False).
+        leds : list
+            List of LED name string to indicate which LEDs to turn on.
+
+        """
+        # pass a list of strings of the leds to turn on
+        if not(self.debug_lights):
+            return None
+        if leds == []:
+            # if leds is an empty list do nothing
+            return None
+
+        for led in leds:
+            if desired_state is 1:
+                self.led_status |= led
+            elif desired_state is 0:
+                self.led_status &= ~led
+        # pass the new binary int to LED controller
+        sn3218.enable_leds(self.led_status)
```

Comment:
I'm kind of surprised this works. It would have through you would have needed either `sn3218.enable_leds(self.led_status.value)` or make `LED_light` an `IntFlag` class instead of `Flag`.

Reply:
yup, it did need the .value have fixed